[Multidomain and symfony] add guide for Multidomain and symfony

groetzner-net opened the pull request:
add guide for Multidomain #991 
add guide for symfony #888

luto commented:
Using multiple domains separately within one uberspace is a core feature of uberspace 7, [documented in the manual](https://manual.uberspace.de/web-documentroot/#additional-documentroots). It's already built-in, there is no need to do this manually, sorry :/

I'll take a look at the symfony part of this in a few days, though, thanks!

groetzner-net commented:
> Using multiple domains separately within one uberspace is a core feature of uberspace 7, [documented in the manual](https://manual.uberspace.de/web-documentroot/#additional-documentroots). It's already built-in, there is no need to do this manually, sorry :/
Yes, you are absolutely right, it is a core feature of uberspace. But for me it wasn`t realy clear how to handle multiple Domains for different projects within one account. So call it an extented walk-through. Maybe someone can benefit from it.

luto commented:
I am not quite sure what you mean. Requests to `example.com` end up in `/var/www/virtual/isabell/example.com` (with a fallback to `.../html`, in case a more specific directory does not exist). To handle different projects within one space, put them into different sub folders underneath `/var/www/virtual/isabell` - that's all there is to it. No hashes, `.htaccess` configuration, or symlinks required.

groetzner-net commented:
> I am not quite sure what you mean. Requests to `example.com` end up in `/var/www/virtual/isabell/example.com` (with a fallback to `.../html`, in case a more specific directory does not exist). To handle different projects within one space, put them into different sub folders underneath `/var/www/virtual/isabell` - that's all there is to it. No hashes, `.htaccess` configuration, or symlinks required.

Ok, now I got it, sorry that was too easy. For this of course no guide is needed. So the guide was to learn to write in RST. To be honest, the function of "RewriteBase /" was not quite clear to me.

groetzner-net commented:
@luto do I have to remove the multidomain-guide or do I have to create a new commit and push it again. Sorry this is my first github-project-merge.

luto commented on source/guide_symfony.rst:
```diff
@@ -0,0 +1,198 @@
+
+::
+
+ [isabell@stardust isabell]$ composer require symfony/apache-pack
```
Are you sure that this should be installed directly into `/var/www/virtual/isabell` instead of the `my_project` directory? This way we end up with two sets of composer files:
```
[oslides@pavo oslides]$ ll
total 8
-rw-rw-r--. 1 oslides oslides   65 Apr 19 09:58 composer.json
-rw-rw-r--. 1 oslides oslides 1652 Apr 19 09:58 composer.lock
drwxr-xr-x. 2 oslides oslides   28 Apr 19 09:40 html
drwxrwxr-x. 9 oslides oslides  185 Apr 19 09:58 my_project
drwxrwxr-x. 4 oslides oslides   57 Apr 19 09:58 vendor
[oslides@pavo my_project]$ ll
total 100
drwxrwxr-x. 2 oslides oslides    21 Apr 19 09:58 bin
-rw-rw-r--. 1 oslides oslides  1372 Apr 19 09:58 composer.json
-rw-rw-r--. 1 oslides oslides 91473 Apr 19 09:58 composer.lock
drwxrwxr-x. 4 oslides oslides   114 Apr 19 09:58 config
drwxrwxr-x. 2 oslides oslides    23 Apr 19 09:58 public
drwxrwxr-x. 3 oslides oslides    42 Apr 19 09:58 src
-rw-rw-r--. 1 oslides oslides  3540 Apr 19 09:58 symfony.lock
drwxrwxrwx. 4 oslides oslides    30 Apr 19 09:58 var
drwxrwxr-x. 6 oslides oslides    79 Apr 19 09:58 vendor
```

luto commented on source/guide_symfony.rst:
```diff
@@ -0,0 +1,198 @@
+    [n] No
+    [a] Yes for all packages, only for the current installation session
+    [p] Yes permanently, never ask again for this project
+    (defaults to n): y
```
Please add a ` :emphasize-lines: X` line here to [show that users need to do something here](https://lab.uberspace.de/guide_tarsnap.html#create-some-keyfiles). Check [the tarsnap guide](https://github.com/Uberspace/lab/commit/38af758acdc1280b40f67adf66f8180988a7fe86#diff-92c49ae3702784d68d084a2a13cd5ae4e96fe7fa58038e1c130b6806bbcc0576R123-R131) for a simple example.

luto commented on source/guide_symfony.rst:
```diff
@@ -0,0 +1,198 @@
+
+::
+
+ [isabell@stardust ~]$ ln -s /var/www/virtual/isabell/my_project/public /var/www/virtual/isabell/html/
```
Use variables and skip the final `/` because the `html` directory is not supposed to exist.
```suggestion
 [isabell@stardust ~]$ ln -s /var/www/virtual/$USER/my_project/public /var/www/virtual/$USER/html
```

luto replied:
Since the `html` directory exists by default, this will create a symlink called "public" inside of it:
```
[oslides@pavo oslides]$ ll html
total 4
-rwxr-xr-x. 1 oslides oslides 1175 Apr 19 09:40 nocontent.html
lrwxrwxrwx. 1 oslides oslides   42 Apr 19 10:02 public -> /var/www/virtual/oslides/my_project/public
```
To fix this, delete the directory first using `rm -f html/nocontent.html; rmdir html`.

luto reviewed (changes requested):
Thanks :tada: I added a few comments, please let me know what you think! Once these are resolved and the multidomain guide has been removed, this is ready for merging.

groetzner-net commented:
> Thanks 🎉 I added a few comments, please let me know what you think! Once these are resolved and the multidomain guide has been removed, this is ready for merging.

The guide is now updated, thanks for you very good suggestions.

luto commented:
Thank you! :tada:
